photon-lib/src/main/scala/com/linkedin/photon/ml/function/L2Regularization.scala

```diff
@@ -119,7 +133,7 @@ trait L2RegularizationDiff extends DiffFunction with L2Regularization {
       normalizationContext: BroadcastWrapper[NormalizationContext]): (Double, Vector[Double]) = {
 
     val (baseValue, baseGradient) = super.calculate(input, coefficients, normalizationContext)
-    val valueWithRegularization = baseValue + l2RegValue(convertToVector(coefficients))
+    val valueWithRegularization = baseValue + l2RegValue(convertToVector(coefficients), interceptIndex)
```

Comment:
We shouldn't need to pass `interceptIndex` as it is a property of the trait, thus can already be accessed from `l2RegValue`.